### PR TITLE
chore(deps): update aptu-coder-core to 0.22, serde-saphyr to 0.0.28, bump anyhow/config

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Dependency updates:
- `aptu-coder-core` 0.20.1 -> 0.22.0: all `lang-*` feature flags removed (obsolete in 0.22), `default-features = false` dropped, Markdown now supported via tree-sitter-md
- `serde-saphyr` 0.0.27 -> 0.0.28
- `anyhow` 1.0.102 -> 1.0.103 (lock bump)
- `config` 0.15.24 -> 0.15.25 (lock bump)

Note: reqwest remains pinned at `=0.12`; migrating to 0.13 requires explicit CryptoProvider initialisation due to both aws-lc-rs and ring being in the tree via octocrab, deferred to a separate PR.

## Test changes

Renamed `test_ast_context_markdown_file_skipped` to `test_ast_context_markdown_file_included`: Markdown is now a supported language in aptu-coder-core 0.22.0 (tree-sitter-md added).

### CODE REVIEW

File: crates/aptu-core/src/ast_context.rs [hunk 1]
```diff
@@ -314,13 +314,13 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_ast_context_markdown_file_skipped() {
+    async fn test_ast_context_markdown_file_included() {
         let files = vec![make_pr_file("README.md")];
         let result = build_ast_context(".", &files).await;
-        // Markdown file should be skipped (edge case: unsupported extension)
+        // Markdown is supported in aptu-coder-core >= 0.22.0 (tree-sitter-md)
         assert!(
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-core/src/ast_context.rs:322]: Fixed. The assertion is now split on the feature flag: with `ast-context` it asserts `!result.is_empty()` (strict, as suggested); without it, it asserts `result.is_empty()` (the stub always returns empty). Both configurations are now validated precisely.

File: crates/aptu-core/src/ast_context.rs [hunk 2]
```diff
@@ -314,13 +314,13 @@ mod tests {
...
         let files = vec![make_pr_file("README.md")];
         let result = build_ast_context(".", &files).await;
-        // Markdown file should be skipped (edge case: unsupported extension)
-        assert!(
-            result.is_empty(),
-            "Markdown file should be skipped (not a supported language)"
-        );
+        // Markdown is supported in aptu-coder-core >= 0.22.0 (tree-sitter-md)
+        #[cfg(feature = "ast-context")]
+        assert!(!result.is_empty(), "Markdown file should be processed and return context");
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-core/src/ast_context.rs]: Fixed in the follow-up commit. Added a second step to the `test` CI job that runs `cargo nextest run -p aptu-core --features ast-context`, ensuring the strict `!result.is_empty()` assertion branch executes on every CI run.
